### PR TITLE
magento/devdocs#8666 Adding instruction - how to stop queue workers.

### LINKED ISSUE #8666: How to stop queue workers for magento commerce?
# General issue

## Description:

Feedback on page: /guides/v2.4/config-guide/deployment/pipeline/example/shared-configuration.html

In the documentation it says "If you use Magento Commerce, stop queue workers. TBD".
but there is no explanation on how it is done and no background information why this has to be done.
There should be a link pointing to an article with more information.

### PR DESCRIPTION
**Pull request by DenisSaltanahmedov**
## Purpose of this pull request
This pull request (PR) adds the example to instruction about how to stop queue consumers.

Fixes #8666 

## Affected DevDocs pages
-  https://devdocs.magento.com/guides/v2.4/config-guide/deployment/pipeline/example/shared-configuration.html

## Links to Magento source code
-  guides/v2.4/config-guide/deployment/pipeline/example/shared-configuration.md

### CODE REVIEW

File: src/_includes/config/split-deploy/example_update-prod.md [hunk 1]
```diff
@@ -13,7 +13,23 @@ To update the production system:
 
    For additional options, such as the ability to set an IP address whitelist, see [`magento maintenance:enable`]({{ page.baseurl }}/install-gde/install/cli/install-cli-subcommands-maint.html).
 
-1. If you use {{site.data.var.ee}}, stop queue workers. TBD
+1. If you use {{site.data.var.ee}}, stop queue workers.
+
+   Add changes to the app/etc/env.php:
```
Comment (BarnyShergold) [inline src/_includes/config/split-deploy/example_update-prod.md:18]:
```suggestion
   Set `cron_run` to `false` in `app/etc/env.php`, e.g.:
```
Explains what is happening.

File: src/_includes/config/split-deploy/example_update-prod.md [hunk 2]
```diff
@@ -13,7 +13,23 @@ To update the production system:
...
+       ]
+   ```
+
+   Update the configuration:
+
+   ```bash
+   php bin/magento app:config:import
+   ```
+
+   And `kill` active consumer processes.
```
Comment (BarnyShergold) [inline src/_includes/config/split-deploy/example_update-prod.md:32]:
```suggestion
Finally, `kill` active consumer processes.
```

**Review by BarnyShergold — CHANGES_REQUESTED** [review]
Few clarity suggestions

**Comment by BarnyShergold** [comment]
@DenisSaltanahmedov we are still waiting for you to make the changes on this PR - are you still working on it?
If it's not done in the next 5 days we will have to consider closing this PR and doing the changes on another.

**Comment by dobooth** [comment]
This include is used in 3 separate files. Let's ensure that these changes are appropriate for all the affect topics.

**Comment by DenisSaltanahmedov** [comment]
@BarnyShergold Sorry I miss the PR
Thanks for your review. I have fixed the PR

**Review by dobooth — CHANGES_REQUESTED** [review]
Sorry...hit the wrong button.

**Comment by DenisSaltanahmedov** [comment]
@dobooth Thanks for your review, I have fixed the recommendation.

**Review by dobooth — APPROVED** [review]

**Comment by dobooth** [comment]
@magento import pr to magento-devdocs/devdocs

**Comment by magento-engcom-team** [comment]
@dobooth the pull request successfully imported.

**Comment by dobooth** [comment]
@magento import pr to magento-devdocs/devdocs

**Comment by magento-engcom-team** [comment]
@dobooth, an error occurred during the Pull Request import.

**Comment by keharper** [comment]
To give an update on this PR. I asked for internal confirmation of these changes, but maybe @DenisSaltanahmedov and @BarnyShergold can provide assistance. Here's what I know:

* The topic at https://devdocs.magento.com/guides/v2.4/config-guide/deployment/pipeline/example/shared-configuration.html was written in 2.2 and probably hasn't been updated much since then
* In 2.3.0 or thereabouts, support for messages was moved from Commerce to Open Source.
* I don't see the string `cron_consumers_runner` in Commerce. I do see it in Open Source: `app/code/Magento/MessageQueue/Model/Cron/ConsumersRunner.php`.

My hunch is that the updates in this PR would have been correct for 2.2, but now the line 

"1. If you use {{site.data.var.ee}}, stop queue workers." 

can be changed to something like

"1. Stop any running queue workers."

I'm not 100% sure. I can only see some details, and not the big picture.

**Comment by hostep** [comment]
@keharper: that's correct, you need to stop queue consumers both in Commerce and Open Source since Magento 2.3.x

I'm however not sure if we should recommend `kill -9 {PID}` (signal: `KILL`) since that's a very dangerous command and might cause data los in the queues (not sure though), we have been using `kill {PID}` (`TERM` is the default signal I believe) for a while now and that works perfectly fine.

Also, be aware that a new command is coming, provided by: https://github.com/magento/magento2/pull/31495: `bin/magento queue:consumers:restart` (which will also run by default with `bin/magento setup:upgrade` by that same PR)
Be aware: this command will only update the poison pill version in the database and that will only cause consumers to restart when a new message is about to be processed. So if no new messages are in the queue, the consumer will still be running without being killed even after that new command gets executed.

Hope this helps a bit.

**Comment by keharper** [comment]
Thanks for the update, @hostep. We'll update the docs with your suggestions. I've also asked the author of 31495 to open a docs PR when the new command features have been merged.

**Comment by keharper** [comment]
@magento import code to magento-devdocs/devdocs

**Review by keharper — APPROVED** [review]

**Pull request merged**